src/core.jl

```diff
@@ -91,36 +132,68 @@ end
 
 # Abstract Container Layers
 """
-    AbstractExplicitContainerLayer{layers} <: AbstractExplicitLayer
+    AbstractExplicitContainerLayer{layers,hasparams,hasstate} <: AbstractExplicitLayer{hasparams,hasstate}
 
 Abstract Container Type for certain Lux Layers. `layers` is a tuple containing fieldnames
 for the layer, and constructs the parameters and states using those.
 
 Users implementing their custom layer can extend the same functions as in
+initialstates(::AbstractRNG, ::AbstractExplicitLayer) = NamedTuple()
 [`AbstractExplicitLayer`](@ref)
 """
-abstract type AbstractExplicitContainerLayer{layers} <: AbstractExplicitLayer end
+abstract type AbstractExplicitContainerLayer{layers, hasparams, hasstate} <:
+              AbstractExplicitLayer{hasparams, hasstate} end
 
 function initialparameters(rng::AbstractRNG,
                            l::AbstractExplicitContainerLayer{layers}) where {layers}
     length(layers) == 1 && return initialparameters(rng, getfield(l, layers[1]))
     return NamedTuple{layers}(initialparameters.(rng, getfield.((l,), layers)))
 end
 
+function initialparameters(::AbstractRNG,
+                           ::AbstractExplicitContainerLayer{layers, false, hasstate}) where {
+                                                                                             layers,
+                                                                                             hasstate
+                                                                                             }
+    return NamedTuple()
+end
+
 function initialstates(rng::AbstractRNG,
                        l::AbstractExplicitContainerLayer{layers}) where {layers}
     length(layers) == 1 && return initialstates(rng, getfield(l, layers[1]))
     return NamedTuple{layers}(initialstates.(rng, getfield.((l,), layers)))
 end
 
+function initialstates(::AbstractRNG,
+                       ::AbstractExplicitContainerLayer{layers, hasparams, false}) where {
+                                                                                          layers,
+                                                                                          hasparams
+                                                                                          }
+    return NamedTuple()
+end
+
 function parameterlength(l::AbstractExplicitContainerLayer{layers}) where {layers}
     return sum(parameterlength, getfield.((l,), layers))
 end
 
+function parameterlength(::AbstractExplicitContainerLayer{layers, false, hasstate}) where {
```

Comment:
unused `hasstate`

Reply:
Fixed in [37f4851](https://github.com/avik-pal/Lux.jl/pull/84/commits/37f48519a731897b9de5771ccb714ff0eea2e013)